### PR TITLE
Add method card ids to exporter

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
## Purpose / Description
Main purpose: having libanki similar to Anki's folder Anki.

When there are more exporter, allows to factorize code

## Approach
As in Anki

## How Has This Been Tested?

gradlew. And I expect it to be tested by travis here. 

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
 import java.io.InputStream;
 import java.util.ArrayList;
 import java.util.HashSet;
+import java.util.List;
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java]: this is the codacy error, trivial to fix

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java]: Corrected. And rebased on master

File: AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java [hunk 2]
```diff
@@ -46,17 +46,33 @@
...
     public Exporter(Collection col, Long did) {
         mCol = col;
         mDid = did;
+        mCids = null;
+    }
+
+    /** card ids of cards in deck self.did if it is set, all ids otherwise. */
+    public Long[] cardIds() {
+        Long[] cids;
+        if (mDid < 0) {
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java]: Before we were checking if `mDid == null`, whereas now we're invoking the comparison operator without checking for null, which will presumably crash with NPE when it is indeed null...? This is a place we have to be extra careful when translating from Python. Could you please double check that it's correct for both branches?

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java]: Nice catch! Sorry I missed that

https://docs.oracle.com/javase/specs/jls/se7/html/jls-5.html#jls-5.1.8

> Unboxing conversion converts expressions of reference type to corresponding expressions of primitive type. Specifically, the following eight conversions are called the unboxing conversions:
> From type Long to type long
> If r is a reference of type Long, then unboxing conversion converts r into r.longValue()
> If r is null, unboxing conversion throws a NullPointerException

File: AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java [hunk 3]
```diff
@@ -46,17 +46,33 @@
...
     protected Collection mCol;
     protected Long mDid;
+    protected long[] mCids;
+    protected int mCount;
 
 
     public Exporter(Collection col) {
         mCol = col;
         mDid = null;
+        mCids = null;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java]: Seems to be unused... What's going on here?

**Comment by Arthur-Milchior** [comment]
@timrae It seems I can't answer to your remark directly. Probably because I did force push a correction. So my remark is here.
And I must admit that 24 days after the commit, I can't recall why I thought those variables would be useful. But as you clearly saw, there are not used anywhere.

File: AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java (hunk 2 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.java]: this was resolved

**Comment by mikehardy** [comment]
@timrae I think all your comments were addressed here

**Pull request merged**